### DISCUSSION TITLE
Block traffic from bots and crawlers

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#137] [Anything Else]
"I see many Crawler and Bots in my page views. Is it possible to exclude those?

Yes, bot detection can be improved significantly. Currently I use an open source solution that only looks at the User-Agent header. I’m planning to add detection based on known bot/crawler IP addresses and referrer spammers."

([imported from the old roadmap](https://plausible.nolt.io/96))

**Comment by julianrubisch** [comment]
fwiw, you could look into how https://github.com/ankane/ahoy does it... I think they defer to JS bot detection a lot

**Comment by metmarkosaric** [comment]
https://udger.com/resources/ip-list

**Comment by ukutaht** [comment]
Some more, maybe better suggestions for where to get IP blocklists:

> The official IP lists from cloud providers are the most reliable, for example https://github.com/nccgroup/cloud_ip_ranges From what you describe something like the botscout list would suit your needs, you can find a copy of it here https://github.com/firehol/blocklist-ipsets

The second one aggregates many smaller lists. We would have to go through each to figure out what exactly they block and and what sort of quality we can expect. If they're not kept up-to-date well enough we might risk blocking legit traffic instead.

**Comment by ukutaht** [comment]
Some implementation notes: https://paraxial.io/blog/cloud-ips

**Comment by metmarkosaric** [comment]
Update: We're now filtering traffic from data centers too. An overview of our bot filtering is here: https://plausible.io/docs/dashboard-faq#does-plausible-exclude-known-bots-and-spam-traffic
